### PR TITLE
[BugFix] Fix throwed NPE in TabletStatMgr when there is no alive node

### LINKED ISSUE #42651: NullPointerException in TabletStatMgr
```
2024-03-15 10:11:45.673+08:00 WARN (tablet stat mgr|44) [TabletStatMgr$CollectTabletStatJob.sendTasks():342] Stop sending tablet stat task for partition ssb.lineorder_flat.13424 because no alive node
2024-03-15 10:11:45.674+08:00 INFO (leaderCheckpointer|123) [GlobalStateMgr.loadHeader():1476] finished to replay header from image
2024-03-15 10:11:45.674+08:00 ERROR (tablet stat mgr|44) [Daemon.run():109] daemon thread got exception. name: tablet stat mgr
java.lang.NullPointerException: null
        at com.starrocks.catalog.TabletStatMgr$CollectTabletStatJob.waitResponse(TabletStatMgr.java:372) ~[starrocks-fe.jar:?]
        at com.starrocks.catalog.TabletStatMgr$CollectTabletStatJob.execute(TabletStatMgr.java:330) ~[starrocks-fe.jar:?]
        at com.starrocks.catalog.TabletStatMgr.updateLakeTableTabletStat(TabletStatMgr.java:281) ~[starrocks-fe.jar:?]
        at com.starrocks.catalog.TabletStatMgr.updateLakeTabletStat(TabletStatMgr.java:218) ~[starrocks-fe.jar:?]
        at com.starrocks.catalog.TabletStatMgr.runAfterCatalogReady(TabletStatMgr.java:97) ~[starrocks-fe.jar:?]
        at com.starrocks.common.util.FrontendDaemon.runOneCycle(FrontendDaemon.java:72) ~[starrocks-fe.jar:?]
        at com.starrocks.common.util.Daemon.run(Daemon.java:107) ~[starrocks-fe.jar:?]
2024-03-15 10:11:45.674+08:00 INFO (leaderCheckpointer|123) [GlobalStateMgr.loadImage():1399] S
```

### StarRocks version (Required)
 - main branch commit 148bd97b85079438bca1ddc8e80dadd5fe69926b
 - shared data mode

### PR DESCRIPTION
**Pull request by sduzh**
Fixes #42651

## What type of PR is this:

- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.2
  - [x] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/catalog/TabletStatMgr.java [hunk 1]
```diff
@@ -369,6 +369,9 @@ private void sendTasks() {
         }
 
         private void waitResponse() {
+            if (responseList == null) {
```
Comment (imay) [inline fe/fe-core/src/main/java/com/starrocks/catalog/TabletStatMgr.java:372]: could you add a comment on when this will be null?

Comment (sduzh) [reply fe/fe-core/src/main/java/com/starrocks/catalog/TabletStatMgr.java:373]: Updated

File: fe/fe-core/src/test/java/com/starrocks/catalog/TabletStatMgrTest.java [hunk 2]
```diff
@@ -413,4 +413,38 @@ public TabletStatResponse get(long timeout, @NotNull TimeUnit unit) {
...
+                return 1000L;
+            }
+            @Mock
+            public ComputeNode chooseNode(LakeTablet tablet) {
+                return null;
+            }
+        };
+
+        TabletStatMgr tabletStatMgr = new TabletStatMgr();
+        Deencapsulation.invoke(tabletStatMgr, "updateLakeTableTabletStat", db, table);
```
Comment (kevincai) [inline fe/fe-core/src/test/java/com/starrocks/catalog/TabletStatMgrTest.java:448]: StarRocksChecker.ExpectNoThrow?

Comment (sduzh) [reply fe/fe-core/src/test/java/com/starrocks/catalog/TabletStatMgrTest.java]: Updated

**Comment by sduzh** [comment]
@Mergifyio rebase

**Comment by sduzh** [comment]
@Mergifyio rebase

**Review by Astralidea — APPROVED** [review]

**Review by gengjun-git — APPROVED** [review]

**Review by HangyuanLiu — APPROVED** [review]

**Pull request merged**